ledger/src/blockstore_processor.rs

```diff
@@ -234,9 +234,31 @@ pub fn execute_batch<'a>(
             pre_commit_callback,
         )?;
 
-    if block_verification {
-        check_block_cost_limits_with_timing(batch, bank, timings, &commit_results)?;
-    }
+    let mut check_block_costs_elapsed = Measure::start("check_block_costs");
+    let tx_costs = if block_verification {
+        // Block verification (including unified scheduler) case;
+        // collect and check transaction costs
+        let tx_costs = get_transaction_costs(bank, &commit_results, batch.sanitized_transactions());
+        check_block_cost_limits(bank, &tx_costs).map(|_| tx_costs)
+    } else {
+        // Unified scheduler block production case;
+        // collect transaction costs if we are recording transaction metadata
+        if record_transaction_meta {
+            Ok(get_transaction_costs(
+                bank,
+                &commit_results,
+                batch.sanitized_transactions(),
+            ))
+        } else {
+            Ok(vec![])
+        }
```

Comment:
can we just make this an `else if` and `else` branch instead of a nested `else` then `if`?

Reply:
Yup, done in https://github.com/anza-xyz/agave/pull/5245/commits/2315a5ffe31d66521a97b6a022c076674587c6cc